### PR TITLE
Upgrade helm version to meet compliance

### PR DESCRIPTION
**Pull request by shubham1172**
Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

# Description

This PR upgrades the version of https://github.com/helm/helm to v3.8.1, which contains compliant dependencies per CNCF guidelines.

Commands run:
```bash
go get -u helm.sh/helm/v3
go mod tidy -go=1.16 && go mod tidy -go=1.17
```

## Issue reference

Please reference the issue this PR will close: https://github.com/dapr/dapr/issues/4219

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* (NA) Created/updated tests
* (NA) Extended the documentation

### CODE REVIEW

**Review by mukundansundar — APPROVED** [review]
lgtm @yaron2  Please take a look at this.

**Review by mukundansundar — CHANGES_REQUESTED** [review]
@shubham1172 Can you take a look at this?

**Comment by shubham1172** [comment]
@mukundansundar, it looks related to https://github.com/kubernetes-sigs/json/issues/8. Let me check on how to get past it.

File: go.mod [hunk 1]
```diff
@@ -7,31 +7,28 @@ require (
...
-	golang.org/x/sys v0.0.0-20211007075335-d3039528d8ac
+	golang.org/x/sys v0.0.0-20211216021012-1d35b9e2eb4e
 	gopkg.in/yaml.v2 v2.4.0
-	helm.sh/helm/v3 v3.5.3
-	k8s.io/api v0.20.2
-	k8s.io/apiextensions-apiserver v0.20.2
-	k8s.io/apimachinery v0.20.2
-	k8s.io/cli-runtime v0.20.2
-	k8s.io/client-go v0.20.2
+	helm.sh/helm/v3 v3.8.0-rc.1.0.20220215042704-bc2f1e08a43a
```
Comment (mukundansundar) [inline go.mod:26]: @shubham1172 Is your commit the only change from the last release (3.8.0)? If not we have changes in the client which are not yet released right?

Comment (shubham1172) [reply go.mod:26]: @mukundansundar, I don't think so. This points to my commit hash, which has a delta of around 27 commits after v3.8.0. 

>If not we have changes in the client which are not yet released right?

Can you please clarify on this part, I did not get it.

**Comment by shubham1172** [comment]
Created an issue with helm/helm to check more on this https://github.com/helm/helm/issues/10683

File: go.mod (hunk 1 above)
Comment (mukundansundar) [reply go.mod:26]: if there are more commits after the 3.8 release than the single commit that you made, we have unreleased code also as part of the client right?

Comment (shubham1172) [reply go.mod:26]: That's true. Their next release is 3.8.1 on March 09, so we can choose to wait as well.

Comment (yaron2) [reply go.mod:26]: Let's wait.

**Review by mukundansundar — APPROVED** [review]
lgtm

@yaron2 Please take a look.

**Review by yaron2 — APPROVED** [review]

**Pull request merged**